ros-controls/ros2_control
# Add ros2 control managers

This PR implements: Updated on 7.10.2020
 
* `ResourceManager` - initial version to mange "SystemHardware" only ([Example 1](https://github.com/ros-controls/roadmap/blob/master/design_drafts/components_architecture_and_urdf_examples.md#1-industrial-robots-with-only-one-interface))
* `ros2_control_node` - generic node to start the ROS2 control robots
* `ControllerManager` - has now `ResourceManager` and supports components which supports. controller manager has multiple callback groups.

To limit amount of code in this PR, only functionalities needed for [Example 1 - Robot](https://github.com/ros-controls/roadmap/blob/master/design_drafts/components_architecture_and_urdf_examples.md#1-industrial-robots-with-only-one-interface) will be implemented in this PR.

Solves #141, #142, and #143

To be done:
- [x] functionalitiy in the `ControllerManagerNewWithManager`
- [x] Use Forward controller in the Example 1
- [x] Registring controllers for components (de-registring should be done in follow up, to keep this PR short as possible)
- [x] Remove `executor::` namespace (I am on Ubutnu 18 with eloquent, therefore needed....)


## How to test: `ros2_control` with components

*UPDATE*: check the [README](https://github.com/ros-controls/ros2_control_demos/blob/add_rrbot_system_position_joints/README.md) in `ros2_control_demos`
